fmralign/metrics.py

```diff
@@ -0,0 +1,184 @@
+import numpy as np
+from scipy.stats import pearsonr
+from sklearn.metrics import r2_score
+
+
+def score_voxelwise(ground_truth, prediction, masker, loss,
+                    multioutput='raw_values'):
+    """
+    Calculates loss function for predicted, ground truth
+    arrays. Supported scores are R2, correlation, and normalized
+    reconstruction error (Bazeille et al., 2019)
+
+    Parameters
+    ----------
+    ground_truth: 3D or 4D Niimg
+        Reference image
+    prediction : 3D or 4D Niimg
+        Same shape as `ground_truth`
+    masker: instance of NiftiMasker or MultiNiftiMasker
+        Masker to be used on ground_truth and prediction. For more information see:
+        http://nilearn.github.io/manipulating_images/masker_objects.html
+    loss : str in ['R2', 'corr', 'n_reconstruction_err']
+        The loss function used in scoring. Default is normalized
+        reconstruction error.
+        'R2' :
+            The R2 distance between source and target arrays.
+            Best possible score is 1.0 and it can be negative (because the
+            model can be arbitrarily worse).
+        'corr' :
+            The correlation between source and target arrays.
+        'n_reconstruction_err' :
+            The normalized reconstruction error. A perfect prediction
+            yields a value of 1.0
+    multioutput: str in ['raw_values', 'uniform_average']
+        Defines aggregating of multiple output scores. Default is raw values.
+        'raw_values' :
+            Returns a full set of scores in case of multioutput input.
+        'uniform_average' :
+            Scores of all outputs are averaged with uniform weight.
+
+    Returns
+    -------
+    score : float or ndarray of floats
+        The score or ndarray of scores if ‘multioutput’ is ‘raw_values’.
+        The worst possible score is arbitrarily set to -1 for all metrics.
+    """
+    X_gt = masker.transform(ground_truth)
+    X_pred = masker.transform(prediction)
+
+    if loss is "R2":
+        score = r2_score(X_gt, X_pred, multioutput=multioutput)
+    elif loss is "n_reconstruction_err":
+        score = normalized_reconstruction_error(
+            X_gt, X_pred, multioutput=multioutput)
+    elif loss is "corr":
+        score = np.array([pearsonr(X_gt[:, vox], X_pred[:, vox])[0]  # pearsonr returns both rho and p
+                          for vox in range(X_pred.shape[1])])
```

Comment:
Related to B.T comment above, multioutput not used

```
if multioutput == "uniform_average":
            score = np.mean(score)
```

should do the job, even though I didn't tested it. It's a pity tests didn't get that, we're not testing uniform_average right ?

Reply:
It doesn't look like CI tests are running at all, currently. Should they be ? pytest is passing locally but I'll add a shape check so the previous commit would have failed !